src/Voron/Data/Containers/Container.cs

```diff
@@ -316,29 +312,280 @@ public string Dump()
+                    offsets.Store(offsetArray);
+                    
+                    var sizes = Vector512.BitwiseAnd(
+                        left: metadataItems, 
+                        right: Vector512.Create(ItemMetadata.SizeMask));
+                    
+                    var byteElements = Vector512.Equals(
+                        left: sizes, 
+                        right: Vector512.Create(ItemMetadata.ByteSizeElement));
+                    
+                    var ushortElements = Vector512.Equals(
+                        left: sizes, 
+                        right: Vector512.Create(ItemMetadata.UshortSizeElement));
+                    
+                    var byteSizedPopulation = byteElements.ExtractMostSignificantBits();
+                    var ushortSizedPopulation = ushortElements.ExtractMostSignificantBits();
+                    var usedPopulation = byteSizedPopulation | ushortSizedPopulation;
+                    var zeroPopulation = Vector512.Equals(sizes, Vector512<ushort>.Zero);
+                    usedItems -= BitOperations.PopCount(zeroPopulation.ExtractMostSignificantBits());
+                    
+                    
+                    //sizes are expected to be between 0 and 31, so it is safe to sum
+                    size += Vector512.Sum(sizes);
+                    size -= ItemMetadata.ByteSizeElement * BitOperations.PopCount(byteSizedPopulation);
+                    size -= ItemMetadata.UshortSizeElement * BitOperations.PopCount(ushortSizedPopulation);
+                    
+#if DEBUG
+                    var lessThanOrEqualUshortElement = Vector512.LessThanOrEqualAll(
+                        left: sizes, 
+                        right: Vector512.Create(ItemMetadata.UshortSizeElement));
+                    Debug.Assert(lessThanOrEqualUshortElement);
+
+                    var greaterThanOrEqualZero = Vector512.GreaterThanOrEqualAll(
+                        left: sizes, 
+                        right: Vector512<ushort>.Zero);
+                    Debug.Assert(greaterThanOrEqualZero);
+#endif
+                    
+                    while (usedPopulation != 0)
+                    {
+                        int currentOffset = BitOperations.TrailingZeroCount(usedPopulation);
+                        
+                        //Sets are disjoint, so if the bit is set in byteMask, we read the value as byte,
+                        //otherwise it is ushort
+                        var sizePtr = (pagePtr + offsetArray[currentOffset]);
+                        size += (byteSizedPopulation & (1UL << currentOffset)) != 0 
+                            ? *sizePtr
+                            : *(ushort*)sizePtr;
+                        
+                        usedPopulation &= (usedPopulation - 1);
+                    }
+                }
+            }
+            
+            if (AdvInstructionSet.IsAcceleratedVector256)
+            {
+                var N = Vector256<ushort>.Count;
```

Comment:
Since all VectorX operations have software fallbacks when the instruction set is not available, we should check the performance of the 512 implementation on 256 lanes architectures and the 128 bits on the 256 architectures. This is probably a complicated request for QA, so we can at least try it on a few different platforms to know for sure if it makes sense to have all this repetition.